mitosheet/mitosheet/saved_analyses/step_upgraders/excel_range_import.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+from typing import Any, Dict, List
+
+
+def upgrade_excel_range_import_1_to_2(step: Dict[str, Any], later_steps: List[Dict[str, Any]]) -> List[Dict[str, Any]]: 
+    """
+    Adds an end condition to 'upper left corner value' excel range imports, that defaults to
+    end_condition: 'first_empty_cell'.
+
+
+    OLD: 
+    {
+        "step_version": 1, 
+        "step_type": "excel_range_import", 
+        "params": {
+            "file_path": str,
+            "sheet_name": str,
+            "range_imports": {
+                "type": 'range' | 'upper left corner value',
+                "df_name": str,
+                "value": str
+            }[]
+        }
+    }
+
+    NEW: 
+    {
+        "step_version": 2, 
+        "step_type": "excel_range_import", 
+        "params": {
+            "file_path": str,
+            "sheet_name": str,
+            "range_imports": {
+                "type": 'range',
+                "df_name": str,
+                "value": str
+            } | {
+                "type": 'upper left corner value',
+                "df_name": str,
+                "value": str,
+                "end_condition": {'type': 'first empty cell'} | {'type': 'bottom left corner value', 'value': Any}
```

Comment:
How come value is `str` for the upper left corner value, but `Any` for the end condition value?

Reply:
Oooo its actually not in the type definition: ExcelRangeEndCondition's value is Optional[str]

Reply:
Yeah. It's the best we can do in MyPy I think.